ds701_book/24-NN-II-Backprop.qmd

```diff
@@ -145,79 +120,84 @@ plt.grid()
+$$
+\mathrm{sigmoid}(\sum_i w_i x_i + b) \hspace{10pt} \textrm{where} \hspace{10pt} \mathrm{sigmoid}(x) = \frac{1}{1 + e^{-x}}
+$$
 
 remind you of?
 
 __Answer__
 
 How about the Logistic Regression model?
 
-$$
-\begin{split}
-P(y=1\mid x) & = \frac{e^{\alpha+\beta x}}{1+e^{\alpha+\beta x}}\\
-             & = \frac{e^{\alpha+\beta x}}{1+e^{\alpha+\beta x}} \left( \frac{e^{-(\alpha+\beta x)}}{e^{-(\alpha+\beta x)}} \right) \\
-             & = \frac{e^{(\alpha+\beta x)-(\alpha+\beta x)}}{e^{-(\alpha+\beta x)}+e^{(\alpha+\beta x)-(\alpha+\beta x)}} \\
-             & = \frac{e^0}{e^{-(\alpha+\beta x)} + e^0} \\
-             & = \frac{1}{1 + e^{-(\alpha+\beta x)}}
-\end{split}
-$$
-
-Which is the Sigmoid with $\alpha = 0$ and $\beta = 1$.
-
 In fact, just like in Logistic Regression, we use the sigmoid function on the last layer of a neural network that is
 doing binary classification to output the probabilities.
 
-> So Logistic Regression is similar to one neuron with a sigmoid activation function.
+::: {.callout-important}
+So Logistic Regression is similar to one neuron with a sigmoid activation
+function.
+:::
 
-```{python}
-#| hide_input: true
-#| slideshow: {slide_type: subslide}
-#| tags: [hide-input]
-alphas = [-4, -8,-12,-20, 0]
-betas = [0.4,0.4,0.6,1, 1]
-x = np.arange(-10,35)
-fig = plt.figure(figsize=(8, 6)) 
-ax = plt.subplot(111)
+## Multi-Layer Perceptron (MLP) or Fully Connected Network (FCN)
 
-for i in range(len(alphas)):
-    a = alphas[i]
-    b = betas[i]
-    y = np.exp(a+b*x)/(1+np.exp(a+b*x))
-    ax.plot(x,y,label=r"$\alpha=%d,$    $\beta=%3.1f$" % (a,b))
 
-ax.tick_params(labelsize=12)
-ax.set_xlabel('x', fontsize = 14)
-ax.set_ylabel('$p(x)$', fontsize = 14)
-ax.legend(loc='center left', bbox_to_anchor=(1, 0.5), prop={'size': 16})
-ax.set_title('Logistic Functions', fontsize = 16);
-```
+![](figs/NN-figs/L24-fcn-dag.png){width="75%"}
 
-### Multi-Layer Perceptron (MLP) or Fully Connected Network (FCN)
-
-<center>
-    
-<img src="figs/NN-figs/neural_net2.jpeg" width="75%">
-    
-</center>
-
-
-From [cs231n](https://cs231n.github.io/convolutional-networks/)
+From [Understanding Deep Learning](https://udlbook.github.io).
 
 Multiple artificial neurons can be acting on the same inputs, in what we call
 a _layer_, and we can have more than one _layer_ until we produce one or more
 outputs.
 
-The example above shows a network with _3 inputs_, two layers of neurons, each
-with 4 neurons, followed by one layer that produces a single value output.
+The example above shows a network with _3 inputs_, three hidden layers of 4, 2 and 3 neurons, respectively, followed by one layer that produces two continuous value outputs.
+
+E.g. a regression problem.
+
+## FCN Matrix Equations
+
+We can express the matrix equations for a $K$ layer network as
+
+$$
+\begin{aligned}
+\mathbf{h}_1 &= \mathbf{a}[\mathbf{\beta}_0 + \mathbf{\Omega}_0 \mathbf{x}] \\
+\mathbf{h}_2 &= \mathbf{a}[\mathbf{\beta}_1 + \mathbf{\Omega}_1 \mathbf{h}_1] \\
```

Comment:
I think the \beta_i should be boldface since they are vectors.

Reply:
I do have them marked as boldface but \beta doesn't appear to render that way. Ideas?

Reply:
For symbols like \beta, you need \boldsymbol{\beta}

Reply:
oh! thanks!